### ISSUE TITLE
Pip not installing spylib dependencies

### ISSUE THREAD

**Issue by qw-in** [SatelCreative/spylib#68]
Pip does not appear to install the dependencies of spylib. Steps to reproduce:

```
mkdir /tmp/spylib-test
cd /tmp/spylib-test

virtualenv env
source env/bin/activate.fish

pip install spylib
python

import spylib
```

Received error:
```
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/tmp/spylib-test/env/lib/python3.9/site-packages/spylib/__init__.py", line 6, in <module>
    from .token import OfflineTokenABC, OnlineTokenABC, Token
  File "/tmp/spylib-test/env/lib/python3.9/site-packages/spylib/token.py", line 8, in <module>
    from httpx import AsyncClient, Response
ModuleNotFoundError: No module named 'httpx'
```

I believe this may be due to one of two things:
- Not exporting a `requirements.txt` file before publishing
- The generated `setup.py` not containing the `install_requires` kwarg

**Comment by hillairet** [comment]
Looking into it. Not the first time it's reported but I thought it was fixed.
Really frustrating. It should just work with poetry.

**Comment by lishanl** [comment]
right, it didn't work anymore even for 0.3... 
It worked when I tested it just a few days ago.. 
was I having some kinda dream? .. 😕

**Comment by hillairet** [comment]
I don't know why it worked actually ... :sweat_smile: 
So the problem is that I thought I could build with poetry, that knows the dependencies, and publish with flit, that doesn't know the dependencies but pushes the metadata nicely to PYPI so we have the info of the package displayed.

I will push 0.3.2 with poetry to make the library works even if it kills the info on the pypi page.

**Comment by hillairet** [comment]
Ok finally all good now. I feel a little silly.
For some reason I was convinced that poetry couldn't do the homepage and other metadata on pypi but it can ... so we don't need flit and the whole process goes through poetry which is much better!

**Comment by qw-in** [comment]
Fixed in spylib 0.3.3. Thanks!
